libs/StatusGoQt/src/StatusGo/Accounts/AccountsAPI.cpp

```diff
@@ -0,0 +1,98 @@
+{
+
+Accounts::MultiAccounts getAccounts() {
+    // or even nicer with a raw string literal
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "accounts_getAccounts"},
+        {"params", json::array()}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+
+    return resultJson.get<CallPrivateRpcResponse>().result;
+}
+
+void generateAccountWithDerivedPath(const QString &hashedPassword, const QString &name, const QColor &color, const QString &emoji,
+                                    const QString &path, const QString &derivedFrom)
+{
+    std::vector<json> params = {hashedPassword, name, color, emoji, path, derivedFrom};
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "accounts_generateAccountWithDerivedPath"},
+        {"params", params}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+}
+
+void addAccountWithMnemonicAndPath(const QString &mnemonic, const QString &hashedPassword, const QString &name,
+                                   const QColor &color, const QString &emoji, const QString &path)
+{
+    std::vector<json> params = {mnemonic, hashedPassword, name, color, emoji, path};
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "accounts_addAccountWithMnemonicAndPath"},
+        {"params", params}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+}
+
+void addAccountWatch(const QString &address, const QString &name, const QColor &color, const QString &emoji)
+{
+    std::vector<json> params = {address, name, color, emoji};
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "accounts_addAccountWatch"},
+        {"params", params}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+}
+
+void deleteAccount(const QString &address)
+{
+    std::vector<json> params = {address};
+    json inputJson = {
+        {"jsonrpc", "2.0"},
+        {"method", "accounts_deleteAccount"},
+        {"params", params}
+    };
+
+    auto result = Utils::statusGoCallPrivateRPC(inputJson.dump().c_str());
+    auto resultJson = json::parse(result);
+    checkPrivateRpcCallResultAndReportError(resultJson);
+}
+
+void deleteMultiaccount(const QString &keyUID, const fs::path &keyStoreDir)
+{
+    // We know go bridge won't misbehave with the input arguments
+    auto result = DeleteMultiaccount(const_cast<char*>(keyUID.toStdString().c_str()), const_cast<char*>(keyStoreDir.string().c_str()));
```

Comment:
All those casts look wrong

Reply:
I think QString::toUtf8() / QString::toLocal8Bit() + .data() is easiest way to get char* from QString

Reply:
Yeah, besides the fact that `c_str()` will likely destroy any UTF-8 characters inside the string

Reply:
> All those casts look wrong

The `status-go` wrapper requires char* for the input strings but they are not used for output. I don't want to push this non-const requirement for input strings all the way up to the presentation layer so the only way I found, for now, is to const_cast and rely on the fact that the referenced memory won't be abused.

> I think QString::toUtf8() / QString::toLocal8Bit() + .data() is easiest way to get char* from QString

From my knowledge toStdString is always calling `toUft8`. The docs say "The Unicode data is converted into 8-bit characters using the [toUtf8](https://doc.qt.io/qt-6/qstring.html#toUtf8)() function". Please see [Qt6.3 docs](https://doc.qt.io/qt-6/qstring.html#toStdString)

> Yeah, besides the fact that `c_str()` will likely destroy any UTF-8 characters inside the string

I'm not aware of `c_str` doing any processing on the underlying memory buffer. Can you please share more info on this?